libs/exavmlib/lib/Process.ex

```diff
@@ -140,12 +142,14 @@ defmodule Process do
 
   """
   @spec sleep(timeout) :: :ok
-  def sleep(timeout)
-      when is_integer(timeout) and timeout >= 0
-      when timeout == :infinity do
+  def sleep(timeout) when is_integer(timeout) and timeout >= 0 do
     receive after: (timeout -> :ok)
   end
 
+  def sleep(:infinity) do
+    receive after: (@max_int -> sleep(:infinity))
```

Comment:
I am confused about why we need `@max_int` here. Can't we wait for `:infinity`? Is it a limitation of the VM?

Reply:
Yes, that was the gist of my comment in https://github.com/atomvm/AtomVM/pull/384

See, specifically, 

https://github.com/atomvm/AtomVM/blob/master/src/libAtomVM/opcodesswitch.h#L1776

Maybe we need a comment here and in the Erlang code to indicate that this is a workaround.  And an issue so we can fix it in the VM.